warden/x/intent/keeper/msg_server_new_intent.go

```diff
@@ -27,6 +28,16 @@ func (k msgServer) NewIntent(goCtx context.Context, msg *types.MsgNewIntent) (*t
 		return nil, err
 	}
 
+	err = k.EventService.EventManager(ctx).Emit(ctx,
+		&intent.EventCreateIntent{
+			IntentId: id,
+			Creator:  msg.Creator,
+		})
+
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
```suggestion
	sdkCtx := sdk.UnwrapSDKContext(ctx)
	err = sdkCtx.EventManager().EmitTypedEvent(
		&intent.EventCreateIntent{
			IntentId: id,
			Creator:  msg.Creator,
		})
	if err != nil {
		return nil, err
	}
```

I think that this is the correct way of accessing the EventManager from inside a MsgHandler (but I haven't tested it myself)

Reply:
I will test both options on localnet